### ISSUE TITLE
Add opt-in rollout at physical-space with residual (tendency) prediction (#412)

### ISSUE THREAD

**Issue by Shaerdan** [alan-turing-institute/icenet-mp#409]
Refer to PR #405 as the original, i believe there are some overlaps, the combining this to #405 requires some re-organising first.

### Problem

Two following two structural properties of the current EncodeProcessDecode forecast loop
potentially make it hard for to beat the persistence baseline  (this becomes an issue when testing AMSE loss #304 ):

1. **lacks identity fallback.** for the model to forecast "no change" it must reproduce its
   input through a downsampling encoder, the processor bottleneck, and an upsampling
   decoder. On a synthetic benchmark a trained `cnn_vit_cnn`-style configuration
   scored active-cell MAE 0.093 at lead 1 where literally copying the input scores
   0.034 — the model is ~2.7× worse than a copy at the one behaviour every forecast
   starts from. Persistence, the baseline to beat, is effectively outside the
   hypothesis space.
2. **unconstrained latent feedback.** the processor's raw latent output is appended
   to its own input window without ever being decoded, re-encoded, or tied to the
   distribution the encoders produce; in multi-input configurations the fed-back
   latent also carries channels for input groups (weather, ocean profiles) whose
   future values are never supervised. The cheapest way for training to control the
   resulting compounding is a map that changes little per step — consistent with
   forecasts whose total change across leads is half or less of reality's.

This is also related to #272 again but is a different mechanism:
it persists with the windowed rollout that resolved #272, the original cycling defect.

### Proposed change (opt-in, defaults untouched)

Three config flags on EncodeProcessDecode:

- `rollout_space: physical` — close the forecast loop in observation space: each
  step encodes the current window of physical frames, takes one processor step,
  decodes, and rolls the updated physical field into the window (re-encoded next
  step). Non-target input groups hold their last observed frame.
- `predict_residual: true` — the decoder output is a signed daily increment:
  `x_{k+1} = mask(clip(x_k + delta_k, 0, 1))`. The identity forecast is
  `delta = 0`, so persistence becomes the zero point of the hypothesis space.
- `zero_init_tendency: true` — zero the final decoder convolution at
  initialisation, so an untrained model reproduces (masked) persistence bit-exactly
  at every lead and training starts on the baseline instead of below it.

With the flags absent the class is byte-for-byte unchanged.

### Tested

On a gated synthetic advection benchmark (built so that the best possible forecast
is sharp and beats persistence 3.9–5.6×), a 2×2 factorial {default rollout, physical
residual rollout} × {Huber, AMSE (#304)} gives, as MAE ratio to persistence at
leads 1/3/5/7 (below 1.0 beats persistence; train / validation / test splits):

| arm | train | validation | test |
|---|---|---|---|
| physical residual + AMSE | 0.95/0.75/0.71/0.66 | 0.99/0.86/0.92/0.95 | 0.99/0.86/0.87/0.88 |
| physical residual + Huber | 1.27/0.72/0.62/0.59 | 1.18/0.78/0.83/0.89 | 1.34/0.82/0.79/0.81 |
| default rollout + AMSE | 1.26/0.87/0.82/0.74 | 1.43/1.03/1.05/1.02 | 1.21/0.89/0.87/0.83 |
| default rollout + Huber | 1.29/0.89/0.83/0.75 | 1.45/1.03/1.04/1.00 | 1.23/0.90/0.88/0.84 |

The physical-residual configurations also keep 96–100 % of the target fine-resolution-band
spectral power under the AMSE loss, versus ~0.30–0.37 for the default rollout under
either loss. The AMSE formulation makes it a bit easier to see where the loss of details come 
from and quantifies it.

### Comparable costs

The original latent rollout is cheap per step and this proposal does not change
that, one full training step on a 432² / 14-lead / 3-group configuration measured
3.3× slower with the physical loop (with these 1-layer encoders; the gap was hidden
by data loading in an end-to-end A100 comparison, but it is real). The following is noted for 
the stvit future works: the physical variant is also strictly Markovian in the physical window, 
it cant carry latent memory between forecast steps. A latent space repair with the same
persistence representability property (maybe a re-encoding consistency term) would
enable this proposal there, this remains open.

### Notes

- One DDP interaction surfaced while testing at scale: the model holds parameters
  that receive no gradient in some configurations (will be uploaded soon), which errors 
  default DDP; strategy=ddp_find_unused_parameters_true is the workaround been used so far.
- Relates to: #272 (rollout), #304 / #390 (AMSE loss issue, note that the loss and this change are
  complementary but separate issues)

**Comment by Shaerdan** [comment]
<img width="929" height="355" alt="Image" src="https://github.com/user-attachments/assets/2a4dd826-ba7a-42be-9c52-2d65f49f7224" />

<img width="931" height="377" alt="Image" src="https://github.com/user-attachments/assets/c41240f1-92e7-418e-bc3b-51654af50661" />

**Comment by Shaerdan** [comment]
https://github.com/user-attachments/assets/f93bf504-afde-4fd6-b04e-94e9c2a40e39

results on a toy example

**Comment by Shaerdan** [comment]
<img width="2370" height="393" alt="Image" src="https://github.com/user-attachments/assets/f098091d-2d88-4a18-8a9b-4c1b72112a84" />

<img width="1885" height="523" alt="Image" src="https://github.com/user-attachments/assets/b921eeb5-365e-4096-a5d0-3c6062053f72" />
<img width="1884" height="974" alt="Image" src="https://github.com/user-attachments/assets/7a7b17af-8dfc-4484-b71a-bcdcf38699db" />
<img width="1884" height="929" alt="Image" src="https://github.com/user-attachments/assets/2e1d68e3-5ae6-4161-a12a-ce83a0fa2ca7" />
